CONTRIBUTING.md

```diff
@@ -0,0 +1,94 @@
+agree to abide by our [code of conduct](CONDUCT.md).
+
+## Getting Started
+
+* If you don't have one already, create a [GitHub account](https://github.com/signup/join).
+
+* Search for an existing issue for your problem or suggestion in the [issue tracker](https://github.com/viash-io/viash/issues). If you find a related issue, please comment there rather than creating a new issue. If none can be found, please [create a new issue](https://github.com/viash-io/viash/issues/new) for your problem or suggestion. Clearly describe your issue, including steps to reproduce when it is a bug, or some justification for a proposed improvement.
+
+## Requirements
+
+* [Java 11](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html) or greater
+
+* [sbt](https://www.scala-sbt.org/)
+
+The following dependencies are required for passing some of the tests.
+
+* [Python 3.10](https://www.python.org/downloads/) or greater
+
+* [R 4.0](https://www.r-project.org/) or greater
+
+* [Nextflow](https://www.nextflow.io/) 22.04.5 or greater
+
+* [Docker](https://www.docker.com/)
+
+## Forking the code
+
+* [Fork](https://github.com/viash-io/viash/#fork-destination-box) the repository on GitHub to make a copy of the repository on your account.
+
+* Clone your fork to your local machine so you can edit the files. Make sure to check out the `develop` branch, since this is the branch where all development happens.
+
+  ```bash
+  git clone https://github.com/your-username/viash.git
+  cd viash
+  git checkout develop
+  ```
+
+* Create a new branch for your changes. It's best to keep your changes separate from the `develop` branch.
+
+## Making changes
+
+* Code Standards: Ensure your code adheres to the [Scala style guide](https://docs.scala-lang.org/style/).
+
+* Write Tests: If you add new features or fix bugs, write tests that cover your changes. Our project uses sbt for testing.
+
+* Run Tests Locally: Before submitting your changes, make sure all tests pass locally. Our GitHub Actions CI pipeline performs tests on different environments, but it's good practice to check everything beforehand.
+
+  ```bash
+  sbt test
+  ```
+
+  If you couldn't install some of the required dependencies, you can skip the tests that require them. For example, to skip all tests requiring Docker and Nextflow:
+
+  ```bash
+  sbt 'testOnly -- -l io.viash.DockerTest -l io.viash.NextflowTest'
+  ```
+
+* Documentation: Update the documentation if your changes require it. This includes both in-code documentation and external documentation like README.md.
+
+## Submitting your changes
+
+* Update the Changelog: Add a new entry to the [CHANGELOG.md](CHANGELOG.md) file that describes your changes. Follow the existing format for entries.
+
+* Commit Your Changes: Once you're happy with your changes, commit them to your branch.
+
+  ```sh
+  git commit -m "Add a brief description of your changes"
```

Comment:
Should we be encouraging single line commit messages? It's generally bad form to do this.

Reply:
Not entirely sure what you mean by this; could you elaborate?

Reply:
Something like this?

```suggestion
* Commit your changes: Once you're happy with your changes, commit them to your branch. A commit message should consist of a short summary of the changes, followed by an empty line and a more detailed description. For trivial changes, the more detailed description can be omitted.

  ```sh
  git commit
```

Reply:
Sounds good to me!